### PR TITLE
Tempest's saddlebags now double as ambulance siren/light

### PR DESCRIPTION
**Pull request by Heroman3003**
Added mlembulance mode to Tempest's saddlebags. When in it, they let out a quiet ambulance siren and flash red/blue colors, changing colors of saddlebags themselves in process.

### CODE REVIEW

**Comment by Wickedtemp** [comment]
<3

**Comment by Screemonster** [comment]
I'mma be honest, having seen a gif of this in action, the colours change a touch too slow for the true effect. Use an animated itemstate. Would give the benefit of allowing the front and rear views to show the saddlebags alternating from side to side which is red or blue.

**Comment by Screemonster** [comment]
As for the concept itself, this is fucking stupid and I love it.

**Comment by Heroman3003** [comment]
The problem with animated colorstate, is that while it would allow for faster switching, it would prevent synchronizing of actual mechanical light with switching of color on the sprite, which is why I went with slower one.

**Comment by Wickedtemp** [comment]
So I know everythings on a freeze, but guys, I think this is worth just merging and updating right now, even if it crashes everything.

**Comment by Sebbe9123** [comment]
Plz no

**Comment by Screemonster** [comment]
plz YES

**Comment by Sebbe9123** [comment]
If everything breaks we'll have no more station tho ;~;

**Comment by Heroman3003** [comment]
But the mlembulance!

**Comment by Wickedtemp** [comment]
I feel like Tempest with the Mlembulance could probably save the server if it crashes. We should do it.

**Comment by Novacat** [comment]
Since the new mob AI stuff has not been merged yet on the master branch, we could probably get away with doing an update with what we have...

**Comment by Heroman3003** [comment]
Considering the bug with stupid z-1 caves will be fixed that way, I'm all for it.

**Comment by SpadesNeil** [comment]
>Since the new mob AI stuff has not been merged yet on the master branch, we could probably get away with doing an update with what we have...

https://github.com/VOREStation/VOREStation/pull/4663 Defeats the purpose of a feature freeze, so no.

File: code/modules/vore/fluffstuff/custom_items_vr.dm [hunk 1]
```diff
@@ -614,6 +614,48 @@
...
+	var/ambulance_last_switch = 0
+	var/ambulance_volume = 15
+
+/obj/item/weapon/storage/backpack/saddlebag/tempest/ui_action_click()
+	ambulance = !(ambulance)
+	if(ambulance)
+		processing_objects.Add(src)
+		item_state = "tempestsaddlebag-amb"
+		icon_state = "tempestbag-amb"
+		if (ismob(src.loc))
```
Comment (kevinz000) [inline code/modules/vore/fluffstuff/custom_items_vr.dm]: just loc, same for below

File: code/modules/vore/fluffstuff/custom_items_vr.dm [hunk 2]
```diff
@@ -614,6 +614,48 @@
 	slowdown = 0
 	taurtype = /datum/sprite_accessory/tail/taur/feline/tempest
 	no_message = "These saddlebags seem to be fitted for someone else, and keep slipping off!"
+	action_button_name = "Toggle Mlembulance Mode"
+	var/ambulance = 0
```
Comment (kevinz000) [inline code/modules/vore/fluffstuff/custom_items_vr.dm]: FALSE

File: code/modules/vore/fluffstuff/custom_items_vr.dm [hunk 3]
```diff
@@ -614,6 +614,48 @@
...
+/obj/item/weapon/storage/backpack/saddlebag/tempest/process()
+	if(!ambulance)
+		processing_objects.Remove(src)
+		return
+	if(world.time - ambulance_last_switch > 15)
+		ambulance_state = !(ambulance_state)
+		var/newlight = "#FF0000"
+		if(ambulance_state)
+			newlight = "#0000FF"
+		if (ismob(src.loc))
```
Comment (kevinz000) [inline code/modules/vore/fluffstuff/custom_items_vr.dm]: loc

File: code/modules/vore/fluffstuff/custom_items_vr.dm [hunk 4]
```diff
@@ -614,6 +614,48 @@
...
+/obj/item/weapon/storage/backpack/saddlebag/tempest/ui_action_click()
+	ambulance = !(ambulance)
+	if(ambulance)
+		processing_objects.Add(src)
+		item_state = "tempestsaddlebag-amb"
+		icon_state = "tempestbag-amb"
+		if (ismob(src.loc))
+			var/mob/M = src.loc
+			M.update_inv_back()
+		ambulance_state = 0
```
Comment (kevinz000) [inline code/modules/vore/fluffstuff/custom_items_vr.dm]: FALSE?

File: code/modules/vore/fluffstuff/custom_items_vr.dm [hunk 5]
```diff
@@ -614,6 +614,48 @@
 	slowdown = 0
 	taurtype = /datum/sprite_accessory/tail/taur/feline/tempest
 	no_message = "These saddlebags seem to be fitted for someone else, and keep slipping off!"
+	action_button_name = "Toggle Mlembulance Mode"
+	var/ambulance = 0
+	var/ambulance_state = 0
```
Comment (kevinz000) [inline code/modules/vore/fluffstuff/custom_items_vr.dm]: FALSE if boolean?

**Comment by Wickedtemp** [comment]
Mlemlem, can bump volume to 25? I tested it with headphones and volume turned up to a point where the belly noises from healbelly are nice and audible and fun to listen to - the mlembulance mode at 25 with volume turned up to that extent isn't even as loud as an opening airlock, so I think it'd be a gud level.

**Comment by Heroman3003** [comment]
> 
> 
> I'm nitpicking.

Your input has been seen and completely ignored in favor of keeping the code consistant with way it is in pretty much entirety of rest of codebase.

**Comment by SpadesNeil** [comment]
I am not in favor of keeping it consistent with bad practices.

**Comment by Screemonster** [comment]
Just because it's "consistent" doesn't mean you shouldn't make it good.
Making new things good will mean that eventually the good way will become the way that is consistent.

**Pull request merged**